torch/lib/c10d/TCPStore.hpp

```diff
@@ -36,29 +70,55 @@ class TCPStoreDaemon {
   void getNumKeysHandler(int socket) const;
   void deleteHandler(int socket);
   void waitHandler(int socket);
+  void watchHandler(int socket);
 
   bool checkKeys(const std::vector<std::string>& keys) const;
+  // Helper function to alerts waiting workers, used in setHandler, getHandler
   void wakeupWaitingClients(const std::string& key);
+  // Helper function used when the key is changed
+  // used in setHandler, addHandler, getHandler, deleteHandler
+  void sendKeyUpdatesToClients(
+      const std::string& key,
+      const enum WatchResponseType& type,
+      std::vector<uint8_t>& oldData,
+      std::vector<uint8_t>& newData);
```

Comment:
Curious, why these functions are protected instead of private? Do we expect any other class inherit from `TCPStoreMasterDaemon`?

Reply:
They were previously set as protected, but I will change them all to private since we don't expect inheritance